crates/aptos-rosetta/src/construction.rs

```diff
@@ -313,101 +288,160 @@ async fn construction_metadata(
         _ => {}
     }
 
-    // If both are present, we skip simulation
-    let (suggested_fee, gas_unit_price, max_gas_amount) =
-        if let (Some(gas_unit_price), Some(max_gas_amount)) = (
-            request.options.gas_price_per_unit,
-            request.options.max_gas_amount,
-        ) {
-            let suggested_fee = Amount::suggested_gas_fee(gas_unit_price.0, max_gas_amount.0);
-            (suggested_fee, gas_unit_price.0, max_gas_amount.0)
+    Ok(internal_operation)
+}
+
+async fn simulate_transaction(
+    rest_client: &aptos_rest_client::Client,
+    chain_id: ChainId,
+    options: &MetadataOptions,
+    internal_operation: &InternalOperation,
+    sequence_number: u64,
+) -> ApiResult<(Amount, u64, u64)> {
```

Comment:
Nit: Return a struct or some newtypes, returning a bunch of the same type is a little spooky.